versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/exceptions/CommitConflictException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.common.exceptions;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.projectnessie.versioned.storage.common.logic.CommitConflict;
+
+public class CommitConflictException extends CommitException {
```

Comment:
Most, if not all, exceptions in this package look more like runtime exceptions. Any reason why you preferred to make them all checked? This choice forces you to introduce things like `SQLRunnableExceptions` which might not have been required were these exceptions unchecked in the first place.

Reply:
Those are checked exceptions, to make callers explicitly aware that these can happen (or not). Similar to `NessieConflictException`